include/kernel.h

```diff
@@ -391,6 +391,17 @@ struct _thread_base {
 		u16_t preempt;
 	};
 
+#ifdef CONFIG_SMP
+	/* True for the per-CPU idle threads */
+	u8_t is_idle;
+
+	/* Non-zero when actively running on a CPU */
+	u8_t active;
```

Comment:
Do we need both `active` and `cpu`? Could `active` be a bitmap, where the nth-bit tells us this thread is running on the nth CPU? This would simplify the thread skipping logic in the scheduler a little bit.

Reply:
That's clever.  Yes, I think.  The "cpu" field might in principle be needed (by a fancier scheduler) to tell the kernel the CPU that a non-active thread was "last" scheduled on, but it isn't right now.